uberfire-extensions/uberfire-wires/uberfire-wires-core/uberfire-wires-core-grids/src/test/java/org/uberfire/ext/wires/core/grids/client/widget/grid/impl/BaseKeyboardOperationTest.java

```diff
@@ -0,0 +1,211 @@
+
+        baseKeyboardOperationSpy = spy(new BaseKeyboardOperation(layer) {
+            @Override
+            public int getKeyCode() {
+                return currentKeyCode;
+            }
+
+            @Override
+            public boolean perform(GridWidget gridWidget, boolean isShiftKeyDown, boolean isControlKeyDown) {
+                return false;
+            }
+        });
+    }
+
+    @Test
+    public void scrollSelectedCellIntoView_NoSelection() {
+        assertFalse(baseKeyboardOperationSpy.scrollSelectedCellIntoView(gridWidget));
+    }
+
+    @Test
+    public void scrollSelectedCellIntoView_NoRenderedInformation() {
+        List<GridData.SelectedCell> selectedCells = Arrays.asList(new GridData.SelectedCell(0,0));
+        when(gridData.getSelectedHeaderCells()).thenReturn(selectedCells);
+        when(baseGridRendererHelperMock.getRenderingInformation()).thenReturn(null);
+        assertFalse(baseKeyboardOperationSpy.scrollSelectedCellIntoView(gridWidget));
+    }
+
+    @Test
+    public void scrollSelectedCellIntoView_HeaderSelected() {
+        when(gridData.getSelectedHeaderCells()).thenReturn(Collections.singletonList(new GridData.SelectedCell(0,0)));
+        assertTrue(baseKeyboardOperationSpy.scrollSelectedCellIntoView(gridWidget));
+        verify(layer, never()).getViewport();
+        verify(viewport, never()).getTransform();
+    }
+
+    @Test
+    public void scrollSelectedCellIntoView_CellSelected() {
+        when(gridData.getSelectedCellsOrigin()).thenReturn(new GridData.SelectedCell(0,0));
+        when(gridData.getRow(0)).thenReturn(gridRow);
+        when(gridRow.getHeight()).thenReturn(30d);
+        assertTrue(baseKeyboardOperationSpy.scrollSelectedCellIntoView(gridWidget));
+        verify(layer, never()).getViewport();
+        verify(viewport, never()).getTransform();
+    }
+
+    @Test
+    public void scrollSelectedCellIntoView_HeaderSelected_WithDeltaXScroll() {
+        when(gridData.getSelectedHeaderCells()).thenReturn(Collections.singletonList(new GridData.SelectedCell(0,0)));
+        when(layer.getVisibleBounds()).thenReturn(new BaseBounds(500, 0, BOUNDS_WIDTH, BOUNDS_HEIGHT));
+        assertTrue(baseKeyboardOperationSpy.scrollSelectedCellIntoView(gridWidget));
+        verify(layer, times(1)).getViewport();
+        verify(viewport, times(1)).getTransform();
+    }
+
+    @Test
+    public void scrollSelectedCellIntoView_CellSelected_WithDeltaXScroll() {
+        when(gridData.getSelectedCellsOrigin()).thenReturn(new GridData.SelectedCell(0,0));
+        when(gridData.getRow(0)).thenReturn(gridRow);
+        when(gridRow.getHeight()).thenReturn(30d);
+        when(layer.getVisibleBounds()).thenReturn(new BaseBounds(500, 0, BOUNDS_WIDTH, BOUNDS_HEIGHT));
+        assertTrue(baseKeyboardOperationSpy.scrollSelectedCellIntoView(gridWidget));
+        verify(layer, times(1)).getViewport();
+        verify(viewport, times(1)).getTransform();
```

Comment:
... you should be able to assert that the `Transform` offset X is correctly set here.